Clean up variables, comments, reduce complexity

jasmussen opened the pull request:
This PR does a few things:

1. It replaces the $item-spacing variable with $grid-size, which is one small step towards a base8 based grid.
2. It cleans up comments in the variables file, fixes a few typos, and clarifies comments.
3. It retires a rarely used variable and makes the SCSS math that used it slightly more clear.

Aside from a good code review and testing, this needs a visual regression sanity check. The $grid-size variable is 8px, whereas $item-spacing was 10px. In my own testing, this caused no issues, but it would still be good to check a bit more in-depth. If some margins are now too small, we can consider a 12px variant, or jump directly to 16px.

jasmussen committed:
Clean up variables, comments, reduce complexity

This PR does a few things:

1. It replaces the $item-spacing variable with $grid-size, which is one small step towards a base8 based grid.
2. It cleans up comments in the variables file, fixes a few typos, and clarifies comments.
3. It retires a rarely used variable and makes the SCSS math that used it slightly more clear.

Aside from a good code review and testing, this needs a visual regression sanity check. The $grid-size variable is 8px, whereas $item-spacing was 10px. In my own testing, this caused no issues, but it would still be good to check a bit more in-depth. If some margins are now too small, we can consider a 12px variant, or jump directly to 16px.

jasmussen committed:
Fix typo.

talldan reviewed (approved):
I did a side-by-side test and didn't notice anything broken.

I did spot that #9814 seems to have regressed, and things are more pronounced on this branch.

Nothing major though, it still looks good. 👍

jasmussen committed:
Address feedback.

jasmussen commented:
Solid feedback.

I rebased, and fixed the visual regression:

![screenshot 2018-10-10 at 10 00 21](https://user-images.githubusercontent.com/1204802/46721496-6dc91480-cc73-11e8-9cba-221ebef5cd60.png)

Although that aspect is no longer fully base-8 gridbased, this is to be expected. The regression here is that the X is a dashicon (base10 grid), and the chevrons are material icons (base8 grid). So I had to make an exception there to make them align.

jasmussen committed:
Fix SCSS.
